pkg/network/node/node.go

```diff
@@ -503,3 +508,23 @@ func (node *OsdnNode) handleDeleteService(obj interface{}) {
 	log.V(5).Infof("Watch %s event for Service %q", watch.Deleted, serv.Name)
 	node.DeleteServiceRules(serv)
 }
+
+func validateNetworkPluginName(networkClient networkclient.Interface, pluginName string) error {
+	// Detect any plugin mismatches between node and master
+	clusterNetwork, err := networkClient.Network().ClusterNetworks().Get(networkapi.ClusterNetworkDefault, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return fmt.Errorf("master has not created a default cluster network, network plugin %q can not start", pluginName)
+	} else if err != nil {
+		return fmt.Errorf("cannot fetch %q cluster network: %v", networkapi.ClusterNetworkDefault, err)
+	}
+
+	if clusterNetwork.PluginName != strings.ToLower(pluginName) {
+		if len(clusterNetwork.PluginName) != 0 {
+			return fmt.Errorf("detected network plugin mismatch between OpenShift node(%q) and master(%q)", pluginName, clusterNetwork.PluginName)
+		} else {
+			// Do not return error in this case
+			glog.Warningf(`either there is network plugin mismatch between OpenShift node(%q) and master or OpenShift master is running an older version where we did not persist plugin name`, pluginName)
```

Comment:
They'd have to be running a 3.2 or earlier master for ClusterNetwork.PluginName to be unset. There's no way we'd support a 3.7 node against a 3.2 master even during an upgrade, right? So we could just drop the inner `if` here now.

Reply:
Yeah, dead.  Will remove.